TACC/Core-CMS
# dev → main (GH-73, GH-66 & FP-1099 & GH-93, GH-142, GH-133)

# Important

This PR is __blocked__ intentionally. It will __not__ be merged, and will _eventually_ be closed.

If you were asked to review this PR, then review these instead:
- https://github.com/TACC/Core-CMS/pull/278
- https://github.com/TACC/Core-CMS/pull/275
- https://github.com/TACC/Core-CMS/pull/280
- https://github.com/TACC/Core-CMS/pull/276





<details>
<summary>Click to reveal previous state of PR. <h1>Previous State</h1></summary>



## Important

The branch on [Frontera Dev](https://dev.fronteraweb.tacc.utexas.edu/) could be for testing #259 or #249.

## Overview

- __Plugins: Blockquote & Offset Content__
  Breakpoint: `2400px`
  Plugins: Sample (tweaked)
    https://github.com/TACC/Core-CMS/pull/257
    https://github.com/TACC/Core-CMS/pull/265
- __Setting: Allow Disabling CMS Search Index__
    https://github.com/TACC/Core-CMS/pull/258
- __Plugins: Picture, Link__
    https://github.com/TACC/Core-CMS/pull/261
- __Plugins: Article List, Article Preview__
    https://github.com/TACC/Core-CMS/pull/264
&nbsp;
- Submodule: Style Updates
    https://github.com/TACC/Core-CMS-Resources/pull/54

### Purpose

- Deploy Frontera Homepage (using mostly plugins) ([dev server](http://dev.fronteraweb.tacc.utexas.edu/))
- Allow Texascale to use `main` branch (current deploy uses `dev`)

## Testing

### Remote (Shortcut) Testing

The `dev` branch is deployed on:

- https://dev.fronteraweb.tacc.utexas.edu/ __*__ (look at article lists on Homepage)
- https://texascale-dev.tacc.utexas.edu/ (create test page __‽__ with Blockquote & Offset plugins)

### Local (Complete) Testing

1. <details><summary>Build for New Styles & Migrate for New Plugins</summary>

    First, `docker exec -it core_cms /bin/bash`. Then:

    1. `python3 manage.py migrate`
    1. `npm ci`
    1. `npm run build`
    1. `python3 manage.py collectstatic --no-input`

    </details>
2. <details><summary>Add Plugins to New Page</summary>

    1. Open webpage.
    2. Add test page. __‽__
    3. Add new plugins:
        - Sample (Greet User)
        - Blockquote
        - Offset Content
            - [nest any plugin you want]
        - News Article List (Static)
            - [nest] News Article Preview (Static)
        - Docs Article List (Static)
            - [nest] Docs Article Preview (Static)
        - Allocs Article List
            - [nest] Allocs Article Preview (Static)
        - Events Article List (Static)
            - [nest] Events Article Preview (Static)

    </details>

3. See PRs in "Overview".

> __*__ During your test, Frontera Dev may be running another branch (such as `task/GH-245…` for #249).
> __‽__ The page on which you add plugins will crash if you load another branch that does not have the plugins.

## Notes

<details>
<summary>Complete Lists of Changes</summary>

- Merged PRs to `dev`: https://github.com/TACC/Core-CMS/pulls?q=is%3Apr+is%3Amerged+base%3Adev+sort%3Acreated-desc+-head%3Atest%2FFP-1099-disabled-auto-search-reindex+
- Direct commits to `dev`: 328efb55, a4ce4c3

</details>



</details>
